### PR TITLE
Lib: Add colors to be aligned with the design

### LINKED ISSUE #103: Reference: Dataviz colors
### Description

This issue is for the moment a reference for the colors to be used in ODS Charts. It will evolve to something more specific with actions later on.

- [ ] TODO: define actions
- [ ] TODO: extract names, colors, etc. from the screenshot to be copyable in this issue as real strings

Screenshot 2024-01-30 at 10 28 16

### PR DESCRIPTION
**Pull request by louismaximepiton**
### Related issues

Closes #103

#### New color values and color sets

- Breaking: `ODSCharts.ODSChartsCategoricalColorsSet.DEFAULT_SUPPORTING_COLORS`. It can be replaced by `ODSCharts.ODSChartsCategoricalColorsSet.DEFAULT` or `ODSCharts.ODSChartsCategoricalColorsSet.SUPPORTING_COLORS`.
- Most of the colors hexadecimal values have been refined and refactored to be more consistent and easier to use. It shouldn't affect your current technical usage as the interface remains the same. Please just double-check the rendering of your charts.
- New color sets defined by Orange Design System for light and dark themes:
  - Categorical to distinguish discrete categories of data that are unrelated to one another.
		- `DARK_CATEGORICAL_COLORS_CATEGORICAL` / `LIGHT_CATEGORICAL_COLORS_CATEGORICAL`
  	- `DARK_CATEGORICAL_COLORS_FUNCTIONAL` / `LIGHT_CATEGORICAL_COLORS_FUNCTIONAL`
  	- `DARK_CATEGORICAL_COLORS_PINK` / `LIGHT_CATEGORICAL_COLORS_PINK`
  	- `DARK_CATEGORICAL_COLORS` / `LIGHT_CATEGORICAL_COLORS`
  	- `DARK_CATEGORICAL_COLORS_YELLOW` / `LIGHT_CATEGORICAL_COLORS_YELLOW`
	- Sequential when data values can be ordered from low to high, where data values should be variations of the same colour hue.
		- `DARK_SEQUENTIAL_COLORS_PINK` / `LIGHT_SEQUENTIAL_COLORS_PINK`
		- `DARK_SEQUENTIAL_COLORS_YELLOW` / `LIGHT_SEQUENTIAL_COLORS_YELLOW`

### Motivation & Context

More colors and flexibility for people.
Realign colors with the design with https://github.com/Orange-OpenSource/ods-charts/pull/106/commits/9bec574dbed2a3968011a61734d4ee32d7025b32.
Proposal to make lighter colors darker on dark mode with https://github.com/Orange-OpenSource/ods-charts/pull/106/commits/cd3f3352d7c4f6c13517d81cf1721ba104dc3c1e. Feel free to separate this commit from the PR.
Proposal to handle axis colors differently (even if I can't spot any visual change) with https://github.com/Orange-OpenSource/ods-charts/pull/106/commits/2dab014aabfe3ea8acb28f0e3943b742d3311ca3. Feel free to separate this commit from the PR.

⚠️ TODO after this PR is merged: 
* Change the name of `categoricalColors`, here are some ideas: 
  * `color` as Apache ECharts does, IMO it can be disturbing for newcomers
  * `odsColor` maybe we should prefix all our parameters
  * `odsGraphColor` to be more understandable

### Types of change

- New feature (non-breaking change which adds functionality)

### Test checklist

Please check that the following tests projects are still working:

- [x] `docs/examples`
- [x] `test/angular-ngx-echarts`
- [x] `test/angular-echarts`
- [x] `test/html`
- [x] `test/react`
- [x] `test/vue`
- [x] `test/examples/bar-line-chart`
- [x] `test/examples/single-line-chart`
- [x] `test/examples/timeseries-chart`

### CODE REVIEW

File: docs/examples/index.js [hunk 1]
```diff
@@ -95,12 +95,16 @@ function generateConfigurator(id) {
...
+                <option value="${ODSCharts.ODSChartsCategoricalColorsSet.DEFAULT}">Default colors</option>
+                <option value="${ODSCharts.ODSChartsCategoricalColorsSet.EMPHASIS}">Emphasis</option>
+                <option value="${ODSCharts.ODSChartsCategoricalColorsSet.SUPPORTING_COLORS}">Supporting colors</option>
                 <option value="${ODSCharts.ODSChartsCategoricalColorsSet.LIGHTER_TINTS}">Lighter tints</option>
                 <option value="${ODSCharts.ODSChartsCategoricalColorsSet.DARKER_TINTS}">Darker tints</option>
                 <option value="${ODSCharts.ODSChartsCategoricalColorsSet.SEQUENTIAL_BLUE}">Blue</option>
                 <option value="${ODSCharts.ODSChartsCategoricalColorsSet.SEQUENTIAL_GREEN}">Green</option>
+                <option value="${ODSCharts.ODSChartsCategoricalColorsSet.SEQUENTIAL_PINK}">Pink</option>
                 <option value="${ODSCharts.ODSChartsCategoricalColorsSet.SEQUENTIAL_PURPLE}">Purple</option>
+                <option value="${ODSCharts.ODSChartsCategoricalColorsSet.SEQUENTIAL_YELLOW}">Yellow</option>
```
Comment (jacques-lebourgeois) [inline docs/examples/index.js:107]: This Merge Request reveal we have a bug on the legend of the number of color is less than the number of curves

By instance ODSCharts.ODSChartsSequentialColorsSet.SEQUENTIAL_BLUE has only 5 colors
image

Comment (jacques-lebourgeois) [reply docs/examples/index.js:107]: Do you prefer I fix it in a separate bug?

**Comment by jacques-lebourgeois** [comment]
Ok to merge this PR and create a technicall one to rename  CategoricalColors and SequentialColors to Colors.

(  ODSChartsColorsSet, ODSChartsColor,  ODSChartsCustomColor, ...)

**Review by jacques-lebourgeois — APPROVED** [review]

**Comment by julien-deramond** [comment]
Before merging it, we need to define/precise the exact rules with the designers.

**Comment by louismaximepiton** [comment]
DRAFT:

/*
hue: color, grayed   |    color/white, gray/white   |  

Blue:
600: #085edb, #4f4f4f     |     5.77, 8.19      |      
500: #237eca, #6b6b6b     |     4.27, 5.32      |      
400: #3e9dd6, #878787     |     3.00, 3.59      |      

Purple:
600: #492191, #3a3a3a     |     10.99, 11.37    |      
500: #6e4aa7, #5f5f5f     |     6.56, 6.38      |      
400: #9373bd, #858585     |     3.86, 3.69      |      

Pink:
600: #bc4d9a, #777777     |     4.50, 4.47      |      
500: #d573bb, #999999     |     3.00, 2.84      |      

Green:
600: #0a6e31, #494949     |     6.38, 9.00      |      
500: #198c51, #636363     |     4.21, 6.00      |      
400: #27a971, #7c7c7c     |     3.00, 4.17      |      

Yellow:
600: #9d6e06, #707070     |     4.50, 4.95      |      
500: #b98f11, #8d8d8d     |     3.00, 3.31      |      
*/

/*
hue: color, grayed   |    color/black, gray/black   |  3.5+  5 < x < 13

Blue:
400: #3e9dd6, #878787     |     6.99, 5.84      |      
300: #4bb4e6, #9a9a9a     |     8.97, 7.46      |      
200: #80ceef, #bababa     |     11.99, 10.82      |      

Purple:
400: #9373bd, #858585     |     5.42, 5.69      |      
300: #a885d8, #999999     |     7.01, 7.37      |      
200: #c1a4e4, #b4b4b4     |     9.69, 10.12      |      

Pink:
500: #d573bb, #999999     |     6.99, 7.37      |      
400: #ff8ad4, #b5b5b5     |     9.83, 10.24      |      
300: #ffb4e6, #d0d0d0     |     12.92, 13.61      |    

Green:
400: #27a971, #7c7c7c     |     6.14, 4.41      |      
300: #50be87, #979797     |     7.94, 6.30      |      
200: #84d5af, #b8b8b8     |     10.63, 9.28      |      

Yellow:
500: #b98f11, #8d8d8d     |     6.14, 5.55      |      
400: #ffb400, #b6b6b6     |     10.33, 9.08      |      
300: #ffd200, #c8c8c8     |     12.69, 11.01      |    
*/

File: src/theme/ods-chart-theme.ts [hunk 2]
```diff
@@ -59,12 +59,12 @@ import { isMainAxis } from './const/main-axis-type.const';
...
  *
- * Default color sets are defined so all the colors are `accessible` against white/dark background and distinguishable for most color-blind people. These sets are not accessible from a monochromacy point of view.
+ * Default and Categorical color sets are defined so all the colors are `accessible` against white/dark background and distinguishable for most color-blind people. These sets are not accessible from a monochromacy point of view.
  *
  * If you use our `sequential` sets, please consider using one color out of two if possible for a greater contrast (for example 0-2-4).
  *
- * - {@link DEFAULT} is the default color set for up to 5 different colors meaning 5 different series.
- * - {@link DEFAULT_LONG} is the default color set for up to 10 different colors grouped by 2. You shouldn't need more in your charts.
+ * - {@link DEFAULT} is the default color set for up to 12 different colors.
+ * - {@link CATEGORICAL} is the default design color set for up to 12 different colors.
```
Comment (jacques-lebourgeois) [inline src/theme/ods-chart-theme.ts:67]: I am not sure users will understand the difference between those two sets ?

What do you think about something like :  
```
 * - {@link DEFAULT} is the default color set for up to 12 different colors, selected to favour contrast differences.
 * - {@link CATEGORICAL} is the set of 12 colours for display by category as defined in the Orange Design System recommendations for dataviz.
```

**Review by jacques-lebourgeois — APPROVED** [review]

**Review by jacques-lebourgeois — APPROVED** [review]

**Comment by louismaximepiton** [comment]
Since last time:
- I removed the `DEFAULT_LONG` color set and replaced it by `DEFAULT`. 
- `CATEGORICAL` color set has been added as well to match design specs. 
- I also updated the wording and the color set order to match the design specs.

**Comment by Franco-Riccitelli** [comment]
The categorical colour values and the colour order look good to me.

**Pull request merged**